### ISSUE TITLE
EWC implementation

### ISSUE THREAD

**Issue by pyun-ram** [GT-RIPL/Continual-Learning-Benchmark#7]
Hi! Thanks for your awesome code! It really helps!
I have a problem with the implementation of EWC in this repo.

According to the paper of DeepMind, EWC adopt the Fisher information matrix to approximate the 
![CodeCogsEqn](https://user-images.githubusercontent.com/43157631/91519254-ab48f080-e924-11ea-8f5c-21123210d31e.gif)

I think the $p(\mathcal D_A|\theta)$ should be selected according to the ground truth label.
If adopting the maximum value of predictions as
https://github.com/GT-RIPL/Continual-Learning-Benchmark/blob/d78b9973b6ec0059b2d2577872db355ae2489f6b/agents/regularization.py#L133
it may have chances to wrongly get the likelihood of a false class when computing the fisher information.

**Comment by yenchanghsu** [comment]
The Fisher information has a form of expected value. The more proper way is to sample the ind from the predicted distribution ([see the comment](https://github.com/GT-RIPL/Continual-Learning-Benchmark/blob/master/agents/regularization.py#L135)). Selecting the max is a simplified sampling.

Selecting the ind based on the ground-truth label is called [empirical Fisher](https://github.com/GT-RIPL/Continual-Learning-Benchmark/blob/master/agents/regularization.py#L139). You can try it by setting `--agent_name EWC_online_empFI` ([see code here](https://github.com/GT-RIPL/Continual-Learning-Benchmark/blob/master/agents/customization.py#L71))

All variants mentioned above are approximations and give a similar empirical performance on MNIST.

**Comment by pyun-ram** [comment]
Thank you sooo much for your prompt reply!
If we do not approximate the fisher information with empirical fisher or sampling,
can we just strictly follow the definition to compute the FIM?
The pseudo-code maybe like:
```
input: logits (shape: [1, # of classes]), network
output: weights
probs <- softmax(logits)
weights<-{}
for cls in range(# of classes):
    network.zero_grad()
    curr_y = log(probs[0, cls])
    curr_y.backward()
    for name, param in network.named_parameters():
         weights[name] += param.grad **2 * probs[0, cls]
return weights
```

**Comment by yenchanghsu** [comment]
Sure, please let us know whether it helps. Thank you!

**Comment by pyun-ram** [comment]
Sure! Thank you sooo much for your help! Then I just close this issue. :)
